app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  validates :email, format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
```

Comment:
You'll want a test to go along with this regex. Alternatively, you could look into gems that provide email validations.

Reply:
Okay, a gem might be helpful since there are a lot of different types. I took that regex from a past project that was tested.